app/com/lynxanalytics/biggraph/frontend_operations/Operations.scala

```diff
@@ -415,6 +415,100 @@ abstract class ProjectOperations(env: SparkFreeEnvironment) extends OperationReg
     val op = graph_operations.CreateStringScalar(data)
     op.result.created
   }
+
+  def checkTypeCollision(project: ProjectEditor, other: ProjectViewer) = {
+    val commonAttributeNames =
+      project.vertexAttributes.keySet & other.vertexAttributes.keySet
+    for (name <- commonAttributeNames) {
+      val a1 = project.vertexAttributes(name)
+      val a2 = other.vertexAttributes(name)
+      assert(
+        a1.typeTag.tpe =:= a2.typeTag.tpe,
+        s"Attribute '$name' has conflicting types in the two projects: " +
+          s"(${a1.typeTag.tpe} and ${a2.typeTag.tpe})")
+    }
+  }
+
+  def mergeInto(project: ProjectEditor, other: ProjectViewer): Unit = {
```

Comment:
Is this just code moving, or anything to see here?

Reply:
> Is this just code moving, or anything to see here?

I can actually move it back now because I ended up not using it. Should I? Or do you like it here?

Reply:
I don't really have a strong preference either way. I think it's fine here, but also fine to leave it as it was.